pkg/server/ca/manager.go

```diff
@@ -422,43 +422,59 @@ func (m *manager) loadKeypairSets(ctx context.Context) error {
 		return err
 	}
 
-	lookupX509CA := func(keyID string) *caX509CA {
-		x509CA := x509CAs[keyID]
-		key := keys[keyID]
-		if x509CA != nil && key != nil && certMatchesKey(x509CA.cert, key) {
-			return x509CA
-		}
-		return nil
-	}
-
-	lookupPublicKey := func(keyID string) *caPublicKey {
-		publicKey := publicKeys[keyID]
-		key := keys[keyID]
-		if publicKey != nil && key != nil && publicKeyEqual(publicKey.publicKey, key) {
-			return publicKey
+	populateKeypairSet := func(kps *keypairSet) bool {
+		kps.Reset()
+
+		x509CA := x509CAs[kps.X509CAKeyID()]
+		x509CAKMKey := keys[kps.X509CAKeyID()]
+		x509CAOK := x509CA != nil && x509CAKMKey != nil && certMatchesKey(x509CA.cert, x509CAKMKey)
+		jwtSigningKey := publicKeys[kps.JWTSignerKeyID()]
+		jwtSigningKMKey := keys[kps.JWTSignerKeyID()]
+		jwtSigningKeyOK := jwtSigningKey != nil && jwtSigningKMKey != nil && publicKeyEqual(jwtSigningKey.publicKey, jwtSigningKMKey)
+
+		if !(x509CAOK && jwtSigningKeyOK) {
+			if x509CA == nil && jwtSigningKey == nil {
+				// this is a normal condition when launching the server for the
+				// first time
+				m.c.Log.Debugf("Manager keypair set %q does not exist", kps.slot)
```

Comment:
I think we can drop the `Manager` prefix on all these messages as the field logger should already have `ca_manager` set in `subsystem_name` field

Reply:
👍